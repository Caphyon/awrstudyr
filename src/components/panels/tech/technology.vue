<template>
  <container>
    <div class="scrollable__container">
      <a class="tech" v-for="tech in techs" :href="tech.url" target="_blank">
        <img v-bind:src=" 'https://www.google.com/s2/favicons?domain=' + tech.url " width="16" height="16" class="text--vcenter">
        <span class="text--vcenter">{{tech.name}}</span>
      </a>
    </div>
  </container>
</template>
<script>
import Vue from "vue";
import Container from "./container";
Vue.component("container", Container);

import TechSection from "./../../sections/tech-section.vue";
export default {
  mixins: [TechSection],
  data() {
    return {
      desc:
        "The Technologies section contains a list of frameworks, libraries and other web tools that we found while analyzing the web page."
    };
  }
};
</script>

<style lang="scss" scoped>
  .tech {
    background-color: rgba(0,0,0,.3);
    padding: .5rem;
    color: #fff;
    text-decoration: none;
    font-weight: 400;
    transition: .1s ease-in;

    &:hover {
      padding-left: 1rem;

      span {
        text-decoration: underline;
      }
    }

    img {
      margin-right: .3rem;
    }
  }
</style>
