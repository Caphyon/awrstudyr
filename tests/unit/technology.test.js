import Vue from "vue";
import { shallowMount, config } from "@vue/test-utils";
import technology from "../../src/components/panels/tech/technology.vue"; // name of your Vue component

// you can add as many elements with the same structure as you want here
let techs = [
  {
    name: "Google Analytics",
    url: "http://www.google.com/analytics/"
  },
  {
    name: "SVG",
    url: "http://www.w3.org/Graphics/SVG/"
  }
];

global.EventBus = new Vue();

describe("technology.vue", () => {
  it("Testing technology component", () => {
    config.methods["allTechnologies"] = () => {};
    const wrapper = shallowMount(technology);

    wrapper.setData({
      techs: techs
    });

    // controll if the component is instance
    expect(wrapper.isVueInstance).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();

    // // check if list exists
    expect(wrapper.find('.scrollable__container').exists()).toBe(true);

    // // check if data exists by passing staticly
    expect(wrapper.find('.tech').exists()).toBe(true);
    expect(wrapper.findAll('span').wrappers.length).toBe(wrapper.vm.$data.techs.length);
    expect(wrapper.findAll('img').wrappers.length).toBe(wrapper.vm.$data.techs.length);
  });

  it("Testing technology component without data", () => {
    config.methods["allTechnologies"] = () => {};
    const wrapper = shallowMount(technology);

    wrapper.setData({
      techs: []
    });

    // controll if the component is instance
    expect(wrapper.isVueInstance).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
  });
});
