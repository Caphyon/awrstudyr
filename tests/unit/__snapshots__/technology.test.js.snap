// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`technology.vue Testing technology component 1`] = `
<container-stub>
  <div
    class="scrollable__container"
  >
    <a
      class="tech"
      href="http://www.google.com/analytics/"
      target="_blank"
    >
      <img
        class="text--vcenter"
        height="16"
        src="https://www.google.com/s2/favicons?domain=http://www.google.com/analytics/"
        width="16"
      />
       
      <span
        class="text--vcenter"
      >
        Google Analytics
      </span>
    </a>
    <a
      class="tech"
      href="http://www.w3.org/Graphics/SVG/"
      target="_blank"
    >
      <img
        class="text--vcenter"
        height="16"
        src="https://www.google.com/s2/favicons?domain=http://www.w3.org/Graphics/SVG/"
        width="16"
      />
       
      <span
        class="text--vcenter"
      >
        SVG
      </span>
    </a>
     
    <!---->
  </div>
</container-stub>
`;

exports[`technology.vue Testing technology component without data 1`] = `
<container-stub>
  <div
    class="scrollable__container"
  >
     
    <!---->
  </div>
</container-stub>
`;
